destination/config.go

```diff
@@ -75,10 +82,31 @@ func (c Config) GetCertificateFingerprint() string {
 	return c.CertificateFingerprint
 }
 
-func (c Config) GetIndex() string {
-	return c.Index
-}
-
 func (c Config) GetType() string {
 	return c.Type
 }
+
+// GetIndex determines the index for each record individually.
+// The function might be returning a static index name.
+// If the index is neither static nor a template, an error is returned.
+func (c Config) GetIndex(r opencdc.Record) (string, error) {
+	// Not a template, i.e. it's a static index name
+	if !strings.Contains(c.Index, "{{") && !strings.Contains(c.Index, "}}") {
+		return c.Index, nil
+	}
+
+	// Try to parse the index
+	t, err := template.New("index").Funcs(sprig.FuncMap()).Parse(c.Index)
```

Comment:
This is not ideal, because we will parse the template for each and every record, and this will have a noticeable impact on performance, we should really try to avoid this and parse it only once in `Open`. Btw, I'm thinking we could figure out the actual index name in the `Destination` struct and pass it to the client in `PrepareCreateOperation` alongside the record.

Reply:
Yes, we can fetch the index name by executing the parsed template and pass it to `PrepareCreateOperation` alongside each record.

We can also fetch the index name in `PrepareCreateOperation` from record's metadata using a function like `getIndexName` like

`func (c *Client) PrepareCreateOperation(item opencdc.Record) (interface{}, interface{}, error) {
	// Prepare metadata
	metadata := bulkRequestActionAndMetadata{
		Create: &bulkRequestCreateAction{
			Index: c.getIndexName(item.Metadata),
		},
	}
}
	
func (c *Client) getIndexName(metadata map[string]string) string {
	indexName, ok := metadata[opencdc.MetadataCollection]
	if !ok {
		return c.cfg.GetIndex()
	}

	return indexName
}`

Which approach you would prefer ?

Reply:
We should apply the parsed template, because it gives the user more flexibility specifying the index. It can be either taken from the record (e.g. `{{ index .Metadata "opencdc.collection" }}`), it can be static (e.g. `my_index`) or a combination of both (e.g. `prefix_{{ index .Metadata "opencdc.collection" }}_postfix`). The point is just that we want to parse the template once and then just evaluate it for every record.